src/commands/dev.ts

```diff
@@ -154,3 +167,15 @@ export const devCommand = new Command()
       process.exit(1);
     }
   });
+
+async function restartServer(
+  server: Server<typeof IncomingMessage, typeof ServerResponse>,
+  deployedUrl: string,
+  serverPort: number,
+): Promise<Server<typeof IncomingMessage, typeof ServerResponse>> {
+  server.close();
+  const { spec } = await fetchAndValidateSpec(deployedUrl);
+  const newServer = await startUIServer(API_CONFIG, spec);
+  await open(`http://localhost:${serverPort}`);
+  return newServer;
+}
```

Comment:
This is not the way to do it almost for sure, you need to hot reload the route which is serving the spec not restart the entire server (unless that doesn't work for some reason)

Reply:
all that logic is inside `startUIServer()` so I don't see how I could do that in a straightforward way.
I could try and break everything down and restructure it for that to work, or even fetch the spec inside the GET /api/config route every time it gets called but I don't think it's worth it for either case.

Reply:
the `open()` call may not be necessary but I couldn't find a way to reload ui components so I just opened a new instance

Reply:
Opening and closing the server is gonna be ultra slow and terrible devx, I think its worth tweaking whatever needs to be tweaked to fix this problem properly